myFitnessTrainer/src/screens/WorkoutSummaryScreen.js

```diff
@@ -1,10 +1,63 @@
+import { useState, useEffect } from "react";
 import { StyleSheet, View, Text } from "react-native";
 import GoToDashboardButton from "../components/GoToDashboardButton";
+import { getAllDocuments, getDocumentId } from "../../databaseFunctions";
+import { auth } from "../../firebaseConfig";
 
 const WorkoutSummaryScreen = () => {
+    const [workoutSessionStats, setWorkoutSessionStats] = useState([]);
+
+    useEffect(() => {
+        getWorkoutStatsSessionFromDB().then((result) => {
+            setWorkoutSessionStats(result);
+            // console.log(workoutSessionStats);
+        });
+    }, []);
+
+    const getWorkoutStatsSessionFromDB = async () => {
+        const userId = await getDocumentId(
+            "users",
+            "email",
+            auth.currentUser.email
+        );
+
+        const allExerciseStats = await getAllDocuments("exercise_history");
+
+        // filter by user AND
+        // get the most recent workout session date of user exercise stats
+        const allUserStats = Object.keys(allExerciseStats).filter(
+            (statId) => allExerciseStats[statId]["user_id"] === userId
+        );
+
+        const allUserStatsDates = allUserStats.map(
+            (statId) => allExerciseStats[statId]["date"]
+        );
+
+        let mostRecentDate;
+        allUserStatsDates.forEach((statDate) => {
+            if (mostRecentDate === undefined) {
+                mostRecentDate = new Date(statDate);
+            } else {
+                mostRecentDate = Math.max(
+                    new Date(mostRecentDate),
+                    new Date(statDate)
+                );
+            }
+        });
+        const mostRecentWorkoutStatsId = allUserStats.filter(
```

Comment:
This shouldn't stop us with pushing through this PR, since I think we'll work on the workout summary screen in future weeks - 

I think the reason this might only return one exercise stat is because when each exercise stat is added to our database, the date is specific to that time (to the seconds) that it is submitted. 

A solution to this could be:
1) to add to our exercise_history a 'workout_day' which can be set to the same 'days_completed' value of our workout plan. That way it's easy to get our history for the specific day we are referring too. 
2) make a new date in the exercise_history by doing this:

```
date = new Date();
dateString = date.toLocaleDateString();
dateForDb = Date(dateString);
```
Then, this way any exercise completed on that same day would hold the same value for the Date, since it is only creating a new data based on the date.toLocaleDateString, such as 5-13-2023, not down to the seconds.